src/CategoricalArrays.jl

```diff
@@ -36,4 +34,74 @@ module CategoricalArrays
     include("recode.jl")
 
     include("deprecated.jl")
+
+    function __init__()
+        @require JSON="682c06a0-de6a-54ab-a142-c8b1cf79cde6" begin
+            #import JSON
+            # JSON of CategoricalValue is JSON of the value it refers to
+            JSON.lower(x::CategoricalValue) = JSON.lower(unwrap(x))
+            DataAPI.defaultarray(::Type{CategoricalValue{T, R}}, N) where {T, R} =
+                CategoricalArray{T, N, R}
+            DataAPI.defaultarray(::Type{Union{CategoricalValue{T, R}, Missing}}, N) where {T, R} =
+                CategoricalArray{Union{T, Missing}, N, R}
```

Comment:
why are these definitions linked to JSON? I would assume they make sense in general.

Reply:
Woops. Unfortunately tests don't catch this as JSON is loaded earlier...